server/db/index.js

```diff
@@ -81,46 +59,65 @@ const getReviews = async (page, count, sort, product_id) => {
   count = count ? Number(count) : 5;
   let sortFunction;
 
+  const output = {
+    product: product_id,
+    page: page,
+    count: count,
+    results: [],
+  };
+
   if (sort === "newest") {
     sortFunction = newestCompare;
   } else if (sort === "helpful") {
     sortFunction = helpfulCompare;
   } else if (sort === "relevant") {
     sortFunction = relevantCompare;
   }
-
   const offset = count * (page - 1);
   const totalResults = offset + count;
   const reviewQuery = Review.find()
-    .where({product_id: product_id})
+    .where({ product_id: product_id })
     .limit(totalResults);
-  const result = await reviewQuery.exec();
+
+  console.time('query');
+
+  let reviewResults = await reviewQuery.lean().exec();
   if (sortFunction) {
-    result.sort(sortFunction);
+    reviewResults.sort(sortFunction);
   }
-  result.splice(0, offset);
-  return result;
+  reviewResults.splice(0, offset);
+  // Remove mongo default _id field
+  reviewResults = reviewResults.map(result => {
+    delete result._id;
+    return result;
+  });
+
+  // Create array of executed queries as Promises
+  const photoResults = await reviewResults.map(async (result, index) => {
+    const photoQuery = ReviewPhoto.find()
```

Comment:
Do you need to run a search for each query even though you create indexes in the schema files? Is there not a constant-time lookup for photo ids available?

Reply:
AFAIK (which may be totally flawed), the index is just a way for the db to organize the rows by a certain field so when you search for something with field X, it doesn't have to traverse every document completely. It can use the index it has by ascending X and just go to what you want and grab it. Not sure if that answered the q